src/_guides/language/effective-dart/usage_migrated.md

```diff
@@ -172,6 +240,93 @@ preferred for three main reasons:
     expression is `null`. With `== true`, you have to think through the boolean
     logic to realize that means that a `null` gets converted to *false*.
 
+**Exception:** Using a null-aware operator on a variable inside a condition does
+not promote the variable to a non-nullable type. If you want the variable to be
+promoted inside the body of the if statement, it may be better to use an
+explicit `!= null` check instead of `?.` followed by `??`:
+
+{:.good}
+<?code-excerpt "usage_good.dart (null-aware-promote)"?>
+{% prettify dart tag=pre+code %}
+int measureMessage(String? message) {
+  if (message != null && message.isNotEmpty) {
+    // message is promoted to String.
+    return message.length;
+  }
+
+  return 0;
+}
+{% endprettify %}
+
+{:.bad}
+<?code-excerpt "usage_bad.dart (null-aware-promote)"?>
+{% prettify dart tag=pre+code %}
+int measureMessage(String? message) {
+  if (message?.isNotEmpty ?? false) {
+    // message is not promoted to String.
+    return message!.length;
+  }
+
+  return 0;
+}
+{% endprettify %}
+
+
+### CONSIDER copying a nullable field to a local variable to enable type promotion.
+
+Checking that a nullable variable is not `null` promotes it to a non-nullable type so that you can then call methods on the variable. Unfortunately, that is only sound for local variables and parameters, so fields and top-level variables are not promoted.
```

Comment:
Do we expect readers to understand that "call methods" also covers reading/writing fields/getters?

Reply:
Reworded.